src/Framework/Bootloader/DebugBootloader.php

```diff
@@ -37,25 +44,50 @@ public function __construct(
      */
     public function init(): void
     {
+        $this->initDefaultConfig();
         $this->addStateCollector(EnvironmentCollector::class);
     }
 
     /**
-     * @psalm-param class-string<StateCollectorInterface>|StateCollectorInterface $collector
+     * @param non-empty-string $key
+     * @param TTag $value
+     */
+    public function addTag(string $key, string|\Stringable|\Closure $value): void
+    {
+        $this->config->modify(DebugConfig::CONFIG, new Append('tags', $key, $value));
+    }
+
+    /**
+     * @psalm-param TCollector $collector
      */
-    public function addStateCollector(string|StateCollectorInterface $collector): void
+    public function addStateCollector(string|StateCollectorInterface|Autowire $collector): void
     {
-        $this->collectors[] = $collector;
+        $this->config->modify(DebugConfig::CONFIG, new Append('collectors', null, $collector));
     }
 
     /**
      * Create state and populate it with collectors.
      */
-    private function state(): StateInterface
+    private function state(DebugConfig $config): StateInterface
     {
         $state = new State();
 
-        foreach ($this->collectors as $collector) {
+        foreach ($config->getTags() as $key => $value) {
+            if ($value instanceof \Closure) {
+                $value = $value(...$this->resolver->resolveArguments(new \ReflectionFunction($value)));
```

Comment:
Why don't you use https://github.com/spiral/framework/blob/master/src/Core/src/InvokerInterface.php#L26 ?

Reply:
Good catch, changed.